mkosi

```diff
@@ -557,6 +557,17 @@ def mount_image(args, workspace, loopdev, root_dev, home_dev, srv_dev, root_read
 
             umount(root)
 
+@complete_step("Assigning hostname")
+def assign_hostname(args, workspace):
+    if not args.hostname:
+        return
+
+    root = os.path.join(workspace, "root")
+    hostname_path = os.path.join(root, "etc/hostname")
+    
+    with open(hostname_path, "w+") as f:
```

Comment:
If an image is used with `systemd-nspawn`/`machinectl`, having static hostname will prevent `systemd` from setting transient hostname to machine name - it will always remain e.g. `localhost` despite the `-M` flag.

If this is by design (@poettering ?), then we might probably need to unset static hostname (i.e. remove `/etc/hostname`) if not specified explicitly. In Fedora and Mageia, `/etc/hostname` for "localhost" is installed by default via the `systemd` RPM package.